docs/DeveloperGuide.md

```diff
@@ -182,7 +182,7 @@ The `Storage` component,
 
 The `Database` component,
 * reads the module information from JSON format to the corresponding `ModuleData` object.
-* depends on some classes in the `Model` component (because the `Database` component's job is to retrieve objects that belong to the `Model`)
+* depends on some classes in the `Model` component (because the `Database` component's job is to retrieve objects that belong to the `Model` that can only be read)
```

Comment:
I think a better way to phrase this might be:

"... to save/retrieve readable objects that ..."